lib/broken_link_report.rb

```diff
@@ -0,0 +1,63 @@
+class BrokenLinkReport
```

Comment:
Maybe consider a module rather than a class? This logic appears quite functional as there's no need to save state to objects.

Reply:
I think it best practice in Ruby to do most of the work at the instance level. At the class/module level most operation are carried out against the class singleton - so there is potential for mutation and threading issues.

Reply:
I've create a class method `call` that I think is the best compromise. It provides a simple interface to provide the basic default report of all the current flows, whilst leaving the instance object available to run fine level reports (on individual flow for example) and testing.

Reply:
If you start writing your code at the module level you lose a lot of the elegance of Ruby. The code becomes simply a collection of functional snippets, and as such plays into Ruby's weaknesses. Elixir is a far better language to use if you want to go that way.

The problem with working at the module level is that all actions happen against the module's class instance singleton. That means you can't store state between methods without running the risk of another process using the same methods to mutate the state in a way you weren't expected. So the only alternative is to pass state into each method.

With a multi-step action, passing the state into each method becomes messy because you have to have an external entity to maintain the state between calls, and to split state data up so that each method only gets what it needs (or has to manage data it doesn't need).

What rapidly happens is that the action or method maintaining state become more and more complicated and difficult to test and debug.

With an instance, you are playing to Ruby's strengths. One method can call another without having to pass the state directly into the method because the instance will maintain state. So it makes it a lot easier to write simple more elegant code. It makes it easier to break processes down into smaller steps allowing the process to be interrogated to check functionality. It makes it easy to comply to good coding practices such as restricting each method to a single action.

Reply:
I've renamed `call` as `for_erb_files_at` and change the functionality so that the path is passed in. I think this addresses you're main concern that the class method lacked meaning.